### DISCUSSION TITLE
How to get ShredOS to detect PERC H710P Mini in Dell PowerEdge server?

### DISCUSSION THREAD

**Discussion by LRSFC-DanJ** [PartialVolume/shredos.x86_64#266] [Q&A]
I'd like to use ShredOS to wipe the drives in a Dell PowerEdge R720 server that is due to be decommissioned. The built in "erase" functions in the RAID controller settings are greyed out hence the idea to use ShredOS, however it doesn't appear to be able to detect the RAID controller.

The RAID controller is a Dell PERC H710P Mini, how do I get ShredOS to detect this?

**Comment by PartialVolume** [comment]
Unless there is some way in the bios or configuration to present the drives as non raid, ahci or SATA/SAS then this might not be possible. Some raid controllers don't support individual drive access and only work in raid mode so I believe. It really depends on the controller and system. Does the raid controller config have such a setting for switch raid mode to ahci, SATA, SAS or no raid?

ShredOS/nwipe is not designed to work directly on a raid array as it needs to access an individual disc not an array, hence why you need to have the server present the drives as individual SATA/SAS drives and not an array.

Maybe over at Dell's forum they can tell you whether this is possible with the H710P hardware.

**Reply by LRSFC-DanJ** [reply]
I did look but unfortunately the H710P does not have the option to present the disks as non-RAID.

For the moment I have been able to wipe the disks using a third party commercial utility Active@ KillDisk which can detect the RAID controller and zero the virtual disks, which will suffice for our purposes.

I had just thought perhaps that there might be some driver module that I could drop in somewhere on the ShredOS USB stick to allow it to be able to detect the hardware.

**Reply by PartialVolume** [reply]
There are some raid drivers, that are not currently installed that might recognise these virtual arrays, however I would really need some hardware to test these on, as nwipe would need to put a warning on the erasure certificate that it couldn't obtain direct access to the drive. In addition the raid controller could hide a faulty drive so it might look like the wipe completed successfully but in fact data still exists on a faulty drive.

Depending on how important the erasure is, if it was me I would probably remove each drive and put them in a system that supports HBA, I.e. no raid and use a hex editor to see what is actually on the disc.

Personally I wouldn't wipe raid drives when they are operating in raid mode because I would also want the wipe program to detect I/O errors not have errors hidden by the raid controller. But of course it depends on your use case as to the importance of the wipe.

**Reply by LRSFC-DanJ** [reply]
I just checked `lspci -v` on the server I'm dealing with now (a different one than the one mentioned in the OP but with a similar RAID card) and it is picking up the adapter on the `megaraid_sas` driver, so it doesn't seem like a missing driver is the issue?

**Reply by LRSFC-DanJ** [reply]
`lsscsi` can see the RAID virtual disks:

![PXL_20241031_133741971](https://github.com/user-attachments/assets/92248cd8-6459-4dd5-9d54-ec0e4b1a0a9b)

These SHOULD be on the system as `/dev/sda`, `/dev/sdb` and `/dev/sdc` and I **can** see entries for those in `/dev` but if I try to use any utility like `hdparm` or `smartctl` or `fdisk` or whatever it tells me the devices don't exist?

**Reply by LRSFC-DanJ** [reply]
The previously mentioned Active@KillDisk is based on TinyCore Linux, so it is definitely possible for a Linux system to recognise and wipe these disks, it would just be great to see this supported in an open source utility like ShredOS rather than having to purchase closed source commercial products.

**Reply by PartialVolume** [reply]
What raid configuration is this system using?

**Reply by LRSFC-DanJ** [reply]
I think `/dev/sda` will be a two-drive mirror (RAID 1), then `/dev/sdb` and `/dev/sdc` would be RAID 5 or 6 arrays.

**Comment by Firminator** [comment]
The PERC H710 was always problematic. You are better off taking the drives out and wipe it from another decommisioned server where the RAID controller is already set in a working JBOD mode or a dedicated wiping box with a generic $30 SAS LSI RAID controller in JBOD (and then wiping up to 4 drives drives when time allows). Or you start the journey of crossflashing to IT (JBOD) mode  which in my experience works. Either way there is some research involved getting this going. There were various giudes out there several years ago how to do the crossflashing. At this time I think https://fohdeesha.com/docs/perc.html is well documented and worth a read if you want to go the crossflashing avenue.
